devicetypes/smartenit/smartelek-evse.src/smartelek-evse.groovy

```diff
@@ -0,0 +1,339 @@
+			log.debug "Pilot Short Circuit"
+			return "Short Circuit"
+		case 20:
+			log.debug "Wrong Supply"
+			return "Wrong Supply"
+		case 21:
+			log.debug "GFCI Failure"
+			return "GFCI Failure"
+		case 22:
+			log.debug "GMI Fault"
+			return "GMI Fault"
+		default:
+			log.debug "Unknown fault"
+			return "Unknown fault"
+	}
+}
+
+def parseChargerStatusValue(val) {
+	log.debug "parseChargerStatusValue: ${val}"
+	switch (val as Integer) {
+		case 0:
+			log.debug "value is Unplugged"
+			return "unplugged"
+			break;
+		case 1:
+			log.debug "value is Plugged In"
+			return "pluggedin"
+			break;
+		case 2:
+			return "pluggedin"
+			break;
+		case 3:
+			log.debug "value is Charging"
+			return "charging"
+			break;
+		case 4:
+			log.debug "value is Fault"
+			return "fault"
+			break;
+		case 5:
+			log.debug "value is Charging Completed"
+			return "chargingcompleted"
+			break;
+		default:
+			return ""
+			break;            
+	}
+}
+
+def on() {
+	log.debug "received on command"
+	zigbee.command(EVSECluster, StartCharging, "", [mfgCode: SmartenitMfrCode])
+}
+
+def off() {
+	log.debug "received off command"
+	zigbee.command(EVSECluster, StopCharging, "", [mfgCode: SmartenitMfrCode])
+}
+
+def stopcharging() {
+	log.debug "sending stopcharging command.."
+	zigbee.command(EVSECluster, StopCharging, "", [mfgCode: SmartenitMfrCode])
+}
+
+def startcharging() {
+	log.debug "sending startcharging command.."
+	zigbee.command(EVSECluster, StartCharging, "", [mfgCode: SmartenitMfrCode])
+}
+
+def refresh() {
+	zigbee.readAttribute(zigbee.SIMPLE_METERING_CLUSTER, MeteringCurrentSummation) +
+	zigbee.readAttribute(zigbee.SIMPLE_METERING_CLUSTER, MeteringInstantDemand) +
+	zigbee.readAttribute(EVSECluster, ChargingStatus, [mfgCode: SmartenitMfrCode]) +
+	zigbee.readAttribute(EVSECluster, ChargerVRMS, [mfgCode: SmartenitMfrCode]) +
+	zigbee.readAttribute(EVSECluster, ChargerSessionSummation, [mfgCode: SmartenitMfrCode]) +
+	zigbee.readAttribute(EVSECluster, ChargerSessionDuration, [mfgCode: SmartenitMfrCode]) +
+	zigbee.readAttribute(EVSECluster, ChargerIRMS, [mfgCode: SmartenitMfrCode]) +
+	zigbee.readAttribute(EVSECluster, ChargerLevel, [mfgCode: SmartenitMfrCode]) + 
+	zigbee.readAttribute(EVSECluster, ChargerMaximumCurrent, [mfgCode: SmartenitMfrCode]) +
+	zigbee.readAttribute(EVSECluster, ChargerFault, [mfgCode: SmartenitMfrCode]) +
+	zigbee.readAttribute(EVSECluster, ChargerSessionPeakCurrent, [mfgCode: SmartenitMfrCode]) +
+	zigbee.readAttribute(EVSECluster, ChargerAutoStart, [mfgCode: SmartenitMfrCode])
+}
+
+def configure() {
+	log.debug "in configure()"
+	return (configureHealthCheck() +
+		zigbee.configureReporting(zigbee.SIMPLE_METERING_CLUSTER, MeteringCurrentSummation, 0x25, 0, 600, 50) +
+		zigbee.configureReporting(zigbee.SIMPLE_METERING_CLUSTER, MeteringInstantDemand, 0x2a, 0, 600, 50) +
+		zigbee.configureReporting(EVSECluster, ChargingStatus, 0x30, 0x0, 0x0, null, [mfgCode: SmartenitMfrCode])
+	)
+}
+
+def configureHealthCheck() {
```

Comment:
Can you only configure health check in this method and not return refresh() commands here ? (call refresh() method separately)

Reply:
Now it looks like refresh() is not called anywhere in this DTH.